packages/block-library/src/legacy-widget/edit/create-control.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * External dependencies
+ */
+import { debounce } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+export default function createControl( {
```

Comment:
I would probably write it as a class but this works too 😆

Reply:
I can never decide lol. I like that not using a class means that the "private" methods are actually not accessible. But then it's definitely silly to create a class-like thing and not use a class.

Reply:
One day...

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

Reply:
We could still wrap the class in a static method so that we can control what we really want to expose.

```js
class Control {
  static create(args) {
    const control = new Control(args);

    return {
      element: control.element,
      destroy: control.destroy,
    };
  }
  // ...
}

const { element, destroy } = Control.create(args);
```